src/cdk/stepper/stepper.ts

```diff
@@ -84,7 +97,8 @@ export class CdkStepper {
   @Input()
   get selectedIndex() { return this._selectedIndex; }
   set selectedIndex(index: number) {
-    if (this._selectedIndex != index) {
+    this._steps.toArray()[this._selectedIndex].interacted = true;
+    if (this._selectedIndex != index && !this._steps.toArray()[index].disabled) {
       this._emitStepperSelectionEvent(index);
```

Comment:
I think we only want to emit a change event if the selected index changes due to user interaction, therefore this should be moved to one of the user event handlers